vscode-trace-webviews/src/trace-explorer/time-range/vscode-time-range-data-widget.tsx

```diff
@@ -0,0 +1,136 @@
+
+const JSONBig = JSONBigConfig({
+    useNativeBigInt: true,
+});
+
+interface TimeRangeDataWidgetProps {
+    data: boolean;
+}
+
+interface PersistedData {
+    activeData: string;
+    experimentDataMap: string;
+}
+
+interface vscode {
+    setState(data: PersistedData): void;
+    getState(): PersistedData;
+}
+
+declare const vscode: vscode;
+
+class TimeRangeDataWidget extends React.Component {
+  private _signalHandler: VsCodeMessageManager;
+  private _reactRef: React.RefObject<ReactTimeRangeDataWidget>;
+//   private restoring: boolean;
+
+  static ID = 'trace-explorer-time-range-data-widget';
+  static LABEL = 'Time Range Data';
+
+  constructor(props: TimeRangeDataWidgetProps) {
+      super(props);
+      this._signalHandler = new VsCodeMessageManager();
+      this._reactRef = React.createRef();
+      
+      window.addEventListener('message', event => {
+
+
+          const { command, data } = event.data;
+
+          if (command === 'restore-view') {
+            this.restoreState();
+            return;
+          }
+
+          const experiment = data.wrapper ? convertSignalExperiment(JSONBig.parse(data.wrapper)) : undefined;
+          const parsedData: TimeRangeUpdatePayload = !experiment ? JSONBig.parse(data) : undefined;
+
+          switch (command) {
+            case VSCODE_MESSAGES.EXPERIMENT_SELECTED:
+                signalManager().fireExperimentSelectedSignal(experiment);
+                break;
+            case VSCODE_MESSAGES.EXPERIMENT_UPDATED:
+                if (experiment) {
+                    signalManager().fireExperimentUpdatedSignal(experiment);
+                }
+                break;
+            case VSCODE_MESSAGES.EXPERIMENT_CLOSED:
+                if (experiment) {
+                    signalManager().fireExperimentClosedSignal(experiment);
+                }
+                break;
+            case VSCODE_MESSAGES.SELECTION_RANGE_UPDATED:
+                if (parsedData) {
+                    signalManager().fireSelectionRangeUpdated(parsedData);
+                }
+                break;
+            case VSCODE_MESSAGES.VIEW_RANGE_UPDATED:
+                if (parsedData) {
+                    signalManager().fireViewRangeUpdated(parsedData);
+                }
+                break;
+          }
+
+          this.storeState();
+
+      });
+  }
+
+
+  componentWillUnmount = (): void => {
+      signalManager().off(Signals.REQUEST_SELECTION_RANGE_CHANGE, this.onRequestSelectionChange);
+  }
+
+  onRequestSelectionChange = (payload: TimeRangeUpdatePayload): void => {
+      this._signalHandler.requestSelectionRangeChange(payload);
+  };
+
+  componentDidMount = (): void => {
+      this._signalHandler.notifyReady();
+      signalManager().on(Signals.REQUEST_SELECTION_RANGE_CHANGE, this.onRequestSelectionChange);
+
+  }
+
+  storeState = (): void => {
+      if (this._reactRef.current) {
+        const { state, experimentDataMap } = this._reactRef.current;
```

Comment:
experimenterror TS2341: Property 'experimentDataMap' is private and only accessible within class 'ReactTimeRangeDataWidget'.

This was tested in Gitpod.

Reply:
This error is happening because this needs requires changes to the theia-trace-extension: https://github.com/eclipse-cdt-cloud/theia-trace-extension/pull/991